airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/documentation/documentation.py

```diff
@@ -0,0 +1,605 @@
+            process.is_alive()
+
+        return errors
+
+    def _run(self, connector: Connector) -> CheckResult:
+        if not connector.documentation_file_path or not connector.documentation_file_path.exists():
+            return self.fail(
+                connector=connector,
+                message="Could not check documentation links as the documentation file is missing.",
+            )
+
+        if not connector.documentation_file_path.read_text().rstrip():
+            return self.fail(
+                connector=connector,
+                message="Documentation file is empty",
+            )
+
+        errors = self.validate_links(connector)
+        if errors:
+            return self.fail(
+                connector=connector,
+                message=f"Connector documentation uses invalid links: {', '.join(errors)}",
+            )
+        return self.pass_(
+            connector=connector,
+            message="Documentation links are valid",
+        )
+
+
+class CheckDocumentationHeadersOrder(CheckDocumentationContent):
+    name = "Connectors documentation headers structure, naming and order follow our guidelines"
+
+    CONNECTOR_SPECIFIC_HEADINGS = "<Connector-specific features>"
+
+    @property
+    def description(self) -> str:
+        ordered_headers = TemplateContent("<CONNECTOR_NAME_FROM_METADATA>").headers_with_tag()
+        not_required_headers = [
+            f"Set up the <CONNECTOR_NAME_FROM_METADATA> connector in Airbyte",
+            "For Airbyte Cloud: *(as subtitle of Set up <CONNECTOR_NAME_FROM_METADATA)>",
+            "For Airbyte Open Source: *(as subtitle of Set up <CONNECTOR_NAME_FROM_METADATA)>",
+            self.CONNECTOR_SPECIFIC_HEADINGS + " (but this headers should be on a right place according to expected order)",
+            "Performance considerations",
+            "Data type map",
+            "Limitations & Troubleshooting",
+            "Tutorials",
+        ]
+
+        return generate_description(
+            "documentation_headers_check_description.md", {"ordered_headers": ordered_headers, "not_required_headers": not_required_headers}
+        )
+
+    def not_required_headers(self, connector_name: str) -> list[str]:
+        not_required = [
+            f"Set up the {connector_name} connector in Airbyte",
+            "For Airbyte Cloud:",
+            "For Airbyte Open Source:",
+            self.CONNECTOR_SPECIFIC_HEADINGS,
+            "Performance considerations",
+            "Data type map",
+            "Limitations & Troubleshooting",
+            "Tutorials",
+        ]
+        return not_required
+
+    def check_headers(self, connector: Connector) -> List[str]:
+        """
+        test_docs_structure gets all top-level headers from source documentation file and check that the order is correct.
+        The order of the headers should follow our standard template https://hackmd.io/Bz75cgATSbm7DjrAqgl4rw.
```

Comment:
```suggestion
        The order of the headers should follow our standard template
```
The "standard template" is the `template/template.md` now isn't it?

Reply:
yes, but it follows https://hackmd.io/Bz75cgATSbm7DjrAqgl4rw. and should be updated if it was changed.